tools/test-tools.js

```diff
@@ -100,14 +101,22 @@ export class TestTools {
     })
   }
 
-  fork (script, args = [], options) {
+  /**
+   * @returns {Promise<{stdout: string, stderr: string, exitCode: number}>
+   */
+  fork (script, args = [], options = {}) {
     return new Promise((resolve, reject) => {
       const finalOptions = {
         cwd: this.cwd,
         stdio: [null, null, null],
         ...options
       }
-      const child = spawn(process.execPath, [script, ...args], finalOptions)
+      const nodeArgs = [
+        '--no-warnings',
+        '--loader',
+        pathToFileURL(path.resolve(__dirname, '..', 'node_modules', 'tsm', 'loader.mjs')).toString()
```

Comment:
What about using requie.resolve or import.meta.resolve

```
/// esm
import.meta.resolve('tsm/loader.mjs')
/// cjs
require.resolve('tsm/loader.mjs')
```

Reply:
require.resolve will resolve commonjs export, import.meta.resolve is experimental feature and will print warnings